### DISCUSSION TITLE
VLC integration

### DISCUSSION THREAD

**Discussion by LeumasN** [SEPIA-Framework/sepia-docs#215] [Q&A]
Hi, I am running Sepia on Ras Pi 4 and an android phone.  I have managed to get music playing locally on the phone with VLC by artist and playlist after some messing about.  I would like to have this music streamed to a Chromecast and had hoped for a "cast to my speaker" type of command.  I guess my question is, is there documentation that describes the VLC integration and what control I have over it?

And secondly, is there a way I can get local weather forecasts without entering into a billing agreement with Google (which I would rather not).  At the moment "weather" is failing due to not knowing my location (though I have set up home address, I guess it is working out from GPS coords)

Thanks, Sepia looks great so far.  Planning on trying Home Assistant next.

Martin

**Comment by fquirin** [comment]
> I would like to have this music streamed to a Chromecast and had hoped for a "cast to my speaker" type of command. I guess my question is, is there documentation that describes the VLC integration and what control I have over it?

Unfortunately the latest Android update broke some of the VLC controls (currently working on a fix), but in general you have the basic controls: start, stop, next, back. You can create your own Android intents (via Teach-UI) but I'm not 100% sure what would be necessary to use a Chromecast. Is that actually a feature of VLC itself?

> is there a way I can get local weather forecasts without entering into a billing agreement with Google (which I would rather not). At the moment "weather" is failing due to not knowing my location (though I have set up home address, I guess it is working out from GPS coords)

Yes, you can add GPS coordinates via `settings page 2 - device location` and then you should be able to say "whats the weather?".
If you say "whats the weather at home?" it will use the "home" location that you can enter via settings page 4. Unfortunately I have not exposed the GPS fields that are usually calculated on the SEPIA Server using the default Geo API. Theoretically you could call the accounts API manually to store GPS data 🤔, I should add some info about it [here](https://github.com/SEPIA-Framework/sepia-docs/blob/master/API/assist-server.md) 😅.
Btw Google is not the only supported Geo Coder, you can use [Graphhopper](https://github.com/SEPIA-Framework/sepia-docs/wiki/API-keys#maps---graphhopper) as well without any credit card nonsense ^^. The free quota is usually enough and it keeps active if you use it  once in a while (they deactivate free accounts at some point if they are not used)🙂 .

> Thanks, Sepia looks great so far. Planning on trying Home Assistant next.

🙂 👍 
Let me know if you need help with Home Assistant, the documentation is still under construction 🙈

**Reply by fquirin** [reply]
Oh btw, if you start with Home Assistant, try the ["Internal HUB"](https://github.com/SEPIA-Framework/sepia-docs/wiki/Smart-Home-Controls#tutorial-home-assistant) first. It is probably easier to set up and you can add devices one by one to better control what will be available to SEPIA.

**Reply by LeumasN** [reply]
Thanks for the replies.  Re Chromecast - VLC has a "Cast" button, I was hoping to be able to leverage that.  Suspect I will have to go the whole Home Automation route.

**Reply by fquirin** [reply]
Just to be sure, we are talking about the VLC Android app right? Because for "normal" VLC I've not yet integrated the API (would like to at some point though).

> Suspect I will have to go the whole Home Automation route

How would that look like? Maybe we can simplify it 🙂

**Comment by LeumasN** [comment]
Android VLC, yes.
